src/data/dataconverter.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABCMeta, abstractmethod
+from utils import I3Extractor
+from utils import load_geospatial_data
+from icecube import dataio
+
+class DataConverter(ABCMeta):
+    def __init__(self, paths, mode, pulsemap):
+        self.paths = paths
+        self._extractor = I3Extractor()
+        self.mode = mode
+        self.pulsemap = pulsemap
+
+    def process_file(self, i3_file, gcd_file, mode, pulsemap):
+        gcd_dict, calibration = load_geospatial_data(gcd_file)
+        i3_file = dataio.I3File(i3_file, "r")
+
+        while i3_file.more():
+            try:
+                frame = i3_file.pop_physics()
+            except: 
+                continue
+            array = self._extractor(frame, mode, pulsemap, gcd_dict, calibration, i3_file)
+            self._save(array)
+
+    @abstractmethod
+    def _processfiles(self):
```

Comment:
This class is indeed in a non-usable state. 

I had trouble getting the functionality I wanted from this general class for the i3dataconverter, so I threw away the inheritance of the general class and started building a class that has the functionality I require. I believe it will be easier to define a general class from which we can inherit from when we have one that actually works, which we do now. 

One of the issues I had was that it is restricting for the general class to require a method that process an entire file. I'd like a method that works on a single frame instead, and that it should be class-specific on how we choose to process the files. All we need to inherit is the way in which each frame is read.


.. just my thoughts.